### PR TITLE
Tests for generated C message types

### PR DESCRIPTION
**Pull request by ernestmc**
Connects to #46

### CODE REVIEW

**Comment by wjwwood** [comment]
The changes so far look good to me, is this ready for review?

**Comment by ernestmc** [comment]
@wjwwood I'm still working on tests for the rest of the message types.

**Comment by wjwwood** [comment]
:ok_hand:

**Comment by ernestmc** [comment]
This is ready for review.
The following tests have been implemented:
* primitive types
* primitive types with default values
* string types
* unbounded arrays
* bounded arrays
* static arrays
* complex sub-message type

Tests with valgrind show no memory leaks:
```
$ valgrind --leak-check=yes ./test_interfaces_c
==27107== Memcheck, a memory error detector
==27107== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==27107== Using Valgrind-3.10.1 and LibVEX; rerun with -h for copyright info
==27107== Command: ./test_interfaces_c
==27107== 
Testing rosidl_generator_c message types...
Testing simple primitive message types...
Testing simple primitives with default values...
Testing string types...
Testing primitives unbounded arrays types...
Testing primitives bounded arrays types...
Testing primitives static arrays types...
Testing nested sub-messages...
All tests were good!
==27107== 
==27107== HEAP SUMMARY:
==27107==     in use at exit: 0 bytes in 0 blocks
==27107==   total heap usage: 42 allocs, 42 frees, 2,549 bytes allocated
==27107== 
==27107== All heap blocks were freed -- no leaks are possible
==27107== 
==27107== For counts of detected and suppressed errors, rerun with: -v
==27107== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by tfoote** [comment]
Triggering CI: 

- http://ci.ros2.org/job/ci_linux/763/
- http://ci.ros2.org/job/ci_osx/643/
- http://ci.ros2.org/job/ci_windows/827 (expected)

**Comment by tfoote** [comment]
@ernestmc there's a few linter warnings most obvious in the linux ci job I linked above. Please take a look at those. If that's required as part of the test there should be a way to disable the linter for that line.

**Comment by ernestmc** [comment]
Fixed multi-line string.
Started CI only on Linux:
* http://ci.ros2.org/job/ci_linux/804/

**Comment by ernestmc** [comment]
CI job had errors because I forgot to take out FastRTPS. Re-triggering:
* http://ci.ros2.org/job/ci_linux/806/

**Comment by dirk-thomas** [comment]
I have rescheduled your build as http://ci.ros2.org/job/ci_linux/808/ to first run a build to fix master. Sorry.

**Comment by ernestmc** [comment]
No problem @dirk-thomas. Thanks for the heads-up.

**Comment by tfoote** [comment]
+1

File: rosidl_generator_c/test/test_interfaces.c [hunk 1]
```diff
@@ -0,0 +1,821 @@
...
+  rosidl_generator_c__msg__Int16 int16_msg;
+  rosidl_generator_c__msg__Int32 int32_msg;
+  rosidl_generator_c__msg__Int64 int64_msg;
+  rosidl_generator_c__msg__Uint8 uint8_msg;
+  rosidl_generator_c__msg__Uint16 uint16_msg;
+  rosidl_generator_c__msg__Uint32 uint32_msg;
+  rosidl_generator_c__msg__Uint64 uint64_msg;
+
+  bool_msg.empty_bool = true;
+  assert(bool_msg.empty_bool == true);
```
Comment (dirk-thomas) [inline rosidl_generator_c/test/test_interfaces.c]: `assert` is not a viable way to test in a unit test. When build in release mode none of these checks is performed.

Comment (ernestmc) [inline rosidl_generator_c/test/test_interfaces.c]: I though the tests were only built in the test mode.

**Pull request merged**